pkg/controller/nodes/subworkflow/launchplan.go

```diff
@@ -17,7 +21,8 @@ import (
 )
 
 type launchPlanHandler struct {
-	launchPlan launchplan.Executor
+	launchPlan     launchplan.Executor
+	recoveryClient recovery.Client
```

Comment:
I'm a bit confused about why do we need the recovery client at this layer... won't that be handled at the Node executor layer?

Reply:
Isn't the reason that we want to call the recover execution endpoint instead in case the child workflow node has failed, so that we can recover a partially failed child workflow?

Reply:
That's what I thought.. but looking at the interface and implementation of recovery client... it retrieves Node executions from admin... I think... I obviously could be missing something since I'm half looking at this and half prepping my bags :-D

Reply:
yes, we retrieve the node execution - which has target metadata of type workflow node, which we can use to fetch the originally-launched child execution